File: fix_dns.js
const fs = require('fs');
const axios = require('axios');
const env = fs.readFileSync('.env', 'utf8').split('\n').reduce((acc, line) => {
    const parts = line.split('=');
    const k = parts.shift();
    if(k) acc[k.trim()] = parts.join('=').trim();
    return acc;
}, {});
const TUNNEL_ID = "679587f0-e109-4dfd-a1de-a949f0004809";
const cnameTarget = TUNNEL_ID + ".cfargotunnel.com";

(async () => {
    try {
        const hostname = "admin-gold-gateway.piltismart.com";
        const headers = {
            "Authorization": `Bearer ${env.CF_API_TOKEN}`,
            "Content-Type": "application/json"
        };
        const listRes = await axios.get(`https://api.cloudflare.com/client/v4/zones/${env.CF_ZONE_ID}/dns_records?name=${hostname}`, {headers});
        const records = listRes.data.result;
        if(records.length > 0) {
            await axios.put(`https://api.cloudflare.com/client/v4/zones/${env.CF_ZONE_ID}/dns_records/${records[0].id}`, {
                type: 'CNAME',
                name: hostname,
                content: cnameTarget,
                proxied: true
            }, {headers});
            console.log("DNS Updated to " + cnameTarget);
        } else {
            console.log("Record not found.");
        }
    } catch(e) { console.error(e.response ? e.response.data : e.message); }
})();

File: fix_dns.py
import os, json, requests

env_file = ".env"
with open(env_file) as f:
    env = dict(line.strip().split("=", 1) for line in f if "=" in line)

token = env.get("CF_API_TOKEN")
zone_id = env.get("CF_ZONE_ID")

# The tunnel ID we found on the server
tunnel_id = "679587f0-e109-4dfd-a1de-a949f0004809"
target = f"{tunnel_id}.cfargotunnel.com"
hostname = "admin-gold-gateway.piltismart.com"

headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}
url = f"https://api.cloudflare.com/client/v4/zones/{zone_id}/dns_records?name={hostname}"

r = requests.get(url, headers=headers)
records = r.json().get("result", [])

if records:
    rec_id = records[0]["id"]
    put_url = f"https://api.cloudflare.com/client/v4/zones/{zone_id}/dns_records/{rec_id}"
    data = {"type": "CNAME", "name": hostname, "content": target, "proxied": True}
    requests.put(put_url, headers=headers, json=data)
    print(f"Updated {hostname} to {target}")
else:
    print("Not found")

